### PR TITLE
Stacked Mutation Calculation for Stacked Mutation diagram

### PR DESCRIPTION
**Pull request by joswinter**
For example if we would try to create a stacked mutation container with as number of buckets, 8, the following structure would be generated:

```
level 1: ▄ ▄ ▄ ▄ ▄ ▄ ▄ ▄
level 2: ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄
level 3: ▄▄▄▄▄▄▄ ▄▄▄▄▄▄▄
level 4: ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
```

### CODE REVIEW

File: src/main/java/nl/tudelft/lifetiles/graph/model/BucketCache.java [hunk 1]
```diff
@@ -55,7 +55,7 @@
      */
     public BucketCache(final int numberBuckets,
             final Graph<SequenceSegment> graph) {
-        this.numberBuckets = numberBuckets;
+        this.numberBuckets = (int) Math.pow(2, Math.ceil(Math.log(numberBuckets) / Math.log(2)));
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/model/BucketCache.java:58]: This looks confusing, could use a comment

File: src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+ * Container including the stacks with the mutation quantity used for drawing
+ * the stacked mutation quantity diagram.
+ *
+ * @author Jos
+ *
+ */
+public class StackedMutationContainer {
+
+    /**
+     * List with the stacked quantity of mutations.
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java:21]: Javadoc expects html. A dl tag would be appropriate here

File: src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+     * Public constructor for stacked mutation container.
+     * Will construct the stacked mutation container by calculating the level in
+     * comparison with the bucketCache.
+     *
+     * @param buckets
+     *            BucketCache to be used to generate the stacked mutation
+     *            container.
+     */
+    public StackedMutationContainer(final BucketCache buckets) {
+        level = (int) Math.ceil(Math.log(buckets.getNumberBuckets())
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java:51]: Duplicate code, see BucketCache

File: src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+     * Insert the bucket into the mutation quantity list.
+     *
+     * @param bucket
+     *            Bucket from bucketCache which needs to be added to the stacked
+     *            mutation container.
+     * @return
+     *         Stacked mutation quantity list for this bucket.
+     */
+    private List<Long> insertBucket(final SortedSet<SequenceSegment> bucket) {
+        ArrayList<Long> list = new ArrayList<Long>();
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java:162]: Use List as type

File: src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+     * @param left
+     *            Left section of the child stacked mutation container.
+     * @param right
+     *            Right section of the child stacked mutation container.
+     * @return merged section for this stacked mutation container.
+     */
+    private List<Long> insertStackedContainer(final List<Long> left,
+            final List<Long> right) {
+        ArrayList<Long> stack = new ArrayList<Long>();
+        stack.add(left.get(0) + right.get(0));
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java:123]: Use a loop for this

File: src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+     *
+     * @param left
+     *            Left section of the child stacked mutation container.
+     * @param right
+     *            Right section of the child stacked mutation container.
+     * @return merged section for this stacked mutation container.
+     */
+    private List<Long> insertStackedContainer(final List<Long> left,
+            final List<Long> right) {
+        ArrayList<Long> stack = new ArrayList<Long>();
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/model/StackedMutationContainer.java:122]: Use List as type

**Comment by albert-smit** [comment]
looks good when mentioned issues are fixed

**Comment by jorenham** [comment]
Coveralls died https://github.com/ProgrammingLife5/LifeTiles/issues/176

**Pull request closed without merging**